modules/scripts/startup-script/examples/install_ansible.sh

```diff
@@ -14,40 +14,117 @@
+get_python_major_version() {
+	python_path=${1:-python}
+	python_major_version=$(${python_path} -c "import sys; print(sys.version_info.major)")
+}
+
+# Returns the python minor version. If provided, it will use the first argument
+# as the python executable, otherwise it will default to simply "python".
+get_python_minor_version() {
+	python_path=${1:-python}
+	python_minor_version=$(${python_path} -c "import sys; print(sys.version_info.minor)")
+}
+
+# Install python3 with the yum package manager. Updates python_path to the
+# newly installed packaged.
+install_python3_yum() {
+	## TODO restrict repos to search through in centos to decrease overhead
+	yum install -y python3
+	python_path=$(rpm -ql python3 | grep 'bin/python3$')
+}
+
+# Install python3 with the apt package manager. Updates python_path to the
+# newly installed packaged.
+install_python3_apt() {
+	apt_wait
+	apt-get install -y python3 python3-distutils
+	python_path=$(which python3)
+}
+
+install_python3() {
+	if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ] ||
+		[ -f /etc/oracle-release ] || [ -f /etc/system-release ]; then
+		install_python3_yum
+	elif [ -f /etc/debian_version ] || grep -qi ubuntu /etc/lsb-release 2>/dev/null ||
+		grep -qi ubuntu /etc/os-release 2>/dev/null; then
+		install_python3_apt
+	else
+		echo "Error: Unsupported Distribution"
+		return 1
+	fi
+}
+
+main() {
+	# Get the python3 executable, or install it if not found
+	get_python_path
+	if [ "${python_path}" = "" ]; then
+		if ! install_python3; then
+			return 1
+		fi
+	fi
+	get_python_major_version "${python_path}"
+	if [ "${python_major_version}" = "2" ]; then
+		if ! install_python3; then
+			return 1
 		fi
-		/usr/bin/python -m pip install virtualenv
-		/usr/bin/python -m virtualenv /usr/local/toolkit
-		/usr/local/toolkit/bin/python -m pip install wheel
-		/usr/local/toolkit/bin/python -m pip install ansible==2.9.27
-		ln -s /usr/local/toolkit/bin/ansible-playbook /usr/bin/ansible-playbook
-	elif [ -f /etc/debian_version ] || grep -qi ubuntu /etc/lsb-release || grep -qi ubuntu /etc/os-release; then
-		echo 'WARNING: unsupported installation of ansible in debian / ubuntu'
-		apt_wait
-		apt-get update
-		DEBIAN_FRONTEND=noninteractive apt-get install -y ansible
+	fi
+	install_python_deps
+
+	# Install and/or upgrade pip
+	get_python_minor_version "${python_path}"
+	if [ "${python_minor_version}" -lt 7 ]; then
+		get_pip_url="https://bootstrap.pypa.io/pip/${python_major_version}.${python_minor_version}/get-pip.py"
 	else
-		echo 'Unsupported distribution'
-		exit 1
+		get_pip_url="https://bootstrap.pypa.io/pip/get-pip.py"
 	fi
-fi
+	curl -Os ${get_pip_url}
+	${python_path} get-pip.py
+
+	# Create pip virtual environment for HPC Toolkit
+	${python_path} -m pip install virtualenv
+	${python_path} -m virtualenv /usr/local/ghpc-venv
+	python_path=/usr/local/ghpc-venv/bin/python3
+
+	# Install ansible
+	${python_path} -m pip install ansible==4.10.0
```

Comment:
I set the variable in the script as a global for now. Having to convert this script to a template will be a bit more complicated and disruptive than I anticipated. That can come in a follow-up PR, I think it would make more sense alongside the PR where we automate adding the install script when ansible-local runners are detected.